codegens/curl/test/unit/convert.test.js

```diff
@@ -1,150 +1,54 @@
-      const propertiesTodelete = ['cookies', 'headersSize', 'startedDateTime', 'clientIPAddress'],
-        headersTodelete = [
-          'accept-encoding',
-          'user-agent',
-          'cf-ray',
-          'x-real-ip',
-          'x-request-id',
-          'kong-request-id',
-          'x-request-start',
-          'connect-time',
-          'x-forwarded-for',
-          'content-type',
-          'content-length',
-          'accept',
-          'total-route-time',
-          'cookie',
-          'kong-cloud-request-id',
-          'cache-control',
-          'postman-token'
-        ];
-      if (result[0]) {
-        propertiesTodelete.forEach(function (property) {
-          delete result[0][property];
-        });
-        if (result[0].headers) {
-          headersTodelete.forEach(function (property) {
-            delete result[0].headers[property];
-          });
-        }
-      }
-      if (result[1]) {
-        propertiesTodelete.forEach(function (property) {
-          delete result[1][property];
-        });
-        if (result[1].headers) {
-          headersTodelete.forEach(function (property) {
-            delete result[1].headers[property];
-          });
-        }
-      }
-
-      expect(result[0]).deep.equal(result[1]);
-    }
-    return done();
-  });
-}
+  getUrlStringfromUrlObject = require('../../lib/util').getUrlStringfromUrlObject;
 
 describe('curl convert function', function () {
   describe('convert for different request types', function () {
-
-    mainCollection.item.forEach(function (item) {
-      it(item.name, function (done) {
-        var request = new sdk.Request(item.request),
-          collection = {
-            item: [
-              {
-                request: request.toJSON()
-              }
-            ]
-          },
-          options = {
-            indentCount: 3,
-            indentType: 'Space',
-            requestTimeout: 200,
-            multiLine: true,
-            followRedirect: true,
-            longFormat: true,
-            silent: true,
-            lineContinuationCharacter: '\\'
-          };
-        convert(request, options, function (error, snippet) {
+    var testConfig = {compileScript: null, runScript: null, fileName: null},
+      options = {
+        indentCount: 3,
+        indentType: 'Space',
+        requestTimeout: 200,
+        multiLine: true,
+        followRedirect: true,
+        longFormat: true,
+        silent: true,
+        lineContinuationCharacter: '\\'
+      };
+    async.waterfall([
+      function (next) {
+        newmanTestUtil.generateSnippet(convert, options, function (error, snippets) {
           if (error) {
             expect.fail(null, null, error);
-            return;
+            return next(error);
           }
-          runSnippet(snippet, collection, done);
+
+          return next(null, snippets);
         });
-      });
-    });
+      },
+      function (snippets, next) {
+        snippets.forEach((item, index) => {
```

Comment:
@umeshp7 If this code is common, can we add proper comments to this code in atleast one language, and point to the commented version in all the other languages?
Also, any way to re-use this too?

Reply:
Yes, I will see if we can re-use this too. Meanwhile I will add a comment in curl.

Reply:
Updated this. 
Now you just need to call ```runNewmanTest(convert, options, testConfig);```